etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/RuleEngine.java

```diff
@@ -0,0 +1,59 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+/** Manages the application of rules loaded from a definitions file using the RuleLoader. */
+public class RuleEngine {
+
+    private static final RuleEngine INSTANCE = new RuleEngine();
```

Comment:
I'm wondering if this will be a rules engine for global configuration? an example would be for validation of fhir format or required fields. So if it is, it would carry the same rules across all api requests. If it is not and we are using it as rules for every client, then the rules will differ from client to client. and each client will have to get a different instance of the engine per api request.

Reply:
The rule engine implementation is flexible enough that we can have either global or partner-specific validations (in a big part because we're using FHIR Path which is very powerful). The way to do this would be using the `conditions` in the validation rule. A validation rule with empty conditions will apply to all fhir messages. A sender or receiver-specific one would define a condition like `Bundle.entry.resource.ofType(MessageHeader).destination.receiver.resolve().identifier.where(value = 'simulated-lab-id').exists()`

Reply:
Ah ok. When we do the work for client rules, we would need to look into the singleton design pattern as each client will need to have a different instance of the class.

Reply:
There won't be a need for an instance per partner. All validation rules will be under the same instance. In the future this could be a database instead of the json file. Or azure blob storage

Reply:
I see, so retrieve the rules from the database that has the client settings. This will allow us to make the settings changes without needing to deploy. I like the sound of that